### PR TITLE
[Hub Generated] Review request for Search to add version preview/1.0

### PR DESCRIPTION
**Pull request by daniel-rocha**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/daniel-rocha/azure-rest-api-specs/dev-maps-Search-1.0...main/).

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify **This is a set of fixes for our Swagger to target a new SDK**
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. **Services are already in production**
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month. **Q4**
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/maps/data-plane/Search/preview/1.0/search.json [hunk 1]
```diff
@@ -143,15 +150,120 @@
...
+      "minItems": 2,
+      "maxItems": 2,
+      "items": {
+        "type": "number",
+        "format": "double"
+      }
     },
     "View": {
       "name": "view",
       "in": "query",
```
Comment (tjprescott) [inline specification/maps/data-plane/Search/preview/1.0/search.json:165]: This should align with the name used in https://github.com/Azure/azure-rest-api-specs/pull/16162. In reality, it seems like this param should be entirely removed from this Swagger and instead referenced from `common.json`

File: specification/maps/data-plane/Search/preview/1.0/search.json [hunk 2]
```diff
@@ -261,18 +373,75 @@
...
+              "value": "POI",
+              "description": "Points of interest"
+            },
+            {
+              "value": "Str",
+              "description": "Streets"
+            },
+            {
+              "value": "Xstr",
+              "description": "Cross Streets (Intersections)"
```
Comment (tjprescott) [inline specification/maps/data-plane/Search/preview/1.0/search.json:413]: You may want short names here to drive the autorest-generated labels.

File: specification/maps/data-plane/Search/preview/1.0/search.json [hunk 3]
```diff
@@ -1249,6 +1442,8 @@
             "in": "body",
             "description": "This represents the geometry for one or more geographical features (parks, state boundary etc.) to search in and should be a GeoJSON compliant type. Please refer to [RFC 7946](https://tools.ietf.org/html/rfc7946) for details.",
             "required": true,
+            "x-ms-client-flatten": true,
+            "x-ms-client-name": "geometryParameter",
```
Comment (tjprescott) [inline specification/maps/data-plane/Search/preview/1.0/search.json:1446]: @johanste your take on `x-ms-client-flatten` for body parameters?

Comment (daniel-rocha) [reply specification/maps/data-plane/Search/preview/1.0/search.json:1446]: This works well but still generates the intermediate model classes (SearchAlongRouteRequestBody, SearchInsideGeometryRequestBody) so the gain is pretty small

File: specification/maps/data-plane/Search/preview/1.0/search.json (hunk 1 above)
Comment (alankashiwa) [reply specification/maps/data-plane/Search/preview/1.0/search.json:165]: I've modified the #16162 to use the exact definition of `View` described here. And the references to `common.json` are also made there.

File: specification/maps/data-plane/Search/preview/1.0/search.json [hunk 4]
```diff
@@ -1551,6 +1754,7 @@
           {
             "name": "searchAddressBatchRequestBody",
             "in": "body",
+            "x-ms-client-flatten": true,
```
Comment (alankashiwa) [inline specification/maps/data-plane/Search/preview/1.0/search.json:1757]: This flattening has caused the `searchAddressBatchRequestBody` parameter to disappear in TS SDK. 
And the `batchItem` will be inside `options` and not a required parameter anymore.
image

File: specification/maps/data-plane/Search/preview/1.0/search.json (hunk 3 above)
Comment (daniel-rocha) [reply specification/maps/data-plane/Search/preview/1.0/search.json:1446]: This caused problem with the TS SDK so I am going to revert it.

**Review by tjprescott — CHANGES_REQUESTED** [review]
General comments:
- let's defer any flattening to convenience layer
- In naming types, favor Result over Response
- Consider moving batch operations into their own operation groups.

**Review by tjprescott — APPROVED** [review]

**Pull request merged**